this reverts following commit:

    From: Josh Poimboeuf <jpoimboe@redhat.com>
    Date: Thu, 14 Jan 2021 16:32:42 -0600
    Subject: objtool: Don't fail the kernel build on fatal errors

    [ Upstream commit 655cf86548a3938538642a6df27dd359e13c86bd ]

    This is basically a revert of commit 644592d32837 ("objtool: Fail the
    kernel build on fatal errors").

    That change turned out to be more trouble than it's worth.  Failing the
    build is an extreme measure which sometimes gets too much attention and
    blocks CI build testing.

    These fatal-type warnings aren't yet as rare as we'd hope, due to the
    ever-increasing matrix of supported toolchains/plugins and their
    fast-changing nature as of late.

    Also, there are more people (and bots) looking for objtool warnings than
    ever before, so even non-fatal warnings aren't likely to be ignored for
    long.

    Suggested-by: Nick Desaulniers <ndesaulniers@google.com>
    Reviewed-by: Miroslav Benes <mbenes@suse.cz>
    Reviewed-by: Nick Desaulniers <ndesaulniers@google.com>
    Reviewed-by: Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>
    Signed-off-by: Josh Poimboeuf <jpoimboe@redhat.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

--- a/tools/objtool/check.c
+++ b/tools/objtool/check.c
@@ -4909,10 +4909,14 @@ int check(struct objtool_file *file)
 	}
 
 out:
-	/*
-	 *  For now, don't fail the kernel build on fatal warnings.  These
-	 *  errors are still fairly common due to the growing matrix of
-	 *  supported toolchains and their recent pace of change.
-	 */
+	if (ret < 0) {
+		/*
+		 *  Fatal error.  The binary is corrupt or otherwise broken in
+		 *  some way, or objtool itself is broken.  Fail the kernel
+		 *  build.
+		 */
+		return ret;
+	}
+
 	return 0;
 }
